### PR TITLE
Edit known networks

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/23207

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
On Windows
Behind a feature flag `BraveWalletEditKnownNetworks`:
- Known networks should not be removable.
- Custom networks overriding known ones should be resettable to known.
- Custom networks overriding known ones should take place of known ones in lists(settings and wallet).

### CODE REVIEW

File: browser/resources/settings/brave_wallet_page/wallet_networks_subpage.js [hunk 1]
```diff
@@ -115,6 +152,23 @@ class SettingsWalletNetworksSubpage extends SettingsWalletNetworksSubpageBase {
         then(success => { this.updateNetworks() })
   }
 
+  onResetActionTapped_(event) {
```
Comment (supermassive) [inline browser/resources/settings/brave_wallet_page/wallet_networks_subpage.js:155]: Do we have any browser or js tests to cover such code?

**Comment by supermassive** [comment]
https://user-images.githubusercontent.com/168356/172529676-d4eb66fa-fb20-4bf1-9c3f-01bae16a12c7.mp4

File: browser/resources/settings/brave_wallet_page/wallet_networks_subpage.js (hunk 1 above)
Comment (yrliou) [reply browser/resources/settings/brave_wallet_page/wallet_networks_subpage.js:155]: Not that I know of yet, good to have for sure but I won't block this PR by it.

File: browser/ui/webui/settings/brave_wallet_handler.cc [hunk 2]
```diff
@@ -53,38 +58,60 @@ void BraveWalletHandler::RegisterMessages() {
...
+  CHECK_EQ(args.size(), 2U);
+  PrefService* prefs = GetPrefs();
+  AllowJavascript();
+  DCHECK(brave_wallet::CustomEthChainExists(prefs, args[1].GetString()));
   brave_wallet::RemoveCustomNetwork(prefs, args[1].GetString());
+  DCHECK(brave_wallet::KnownEthChainExists(args[1].GetString()));
   ResolveJavascriptCallback(args[0], base::Value(true));
 }
 
 void BraveWalletHandler::GetCustomNetworksList(const base::Value::List& args) {
```
Comment (yrliou) [inline browser/ui/webui/settings/brave_wallet_handler.cc:84]: This function name should probably be updated to GetNetworkLists.

File: browser/ui/webui/settings/brave_wallet_handler.cc [hunk 3]
```diff
@@ -53,38 +58,60 @@ void BraveWalletHandler::RegisterMessages() {
...
 
 void BraveWalletHandler::RemoveEthereumChain(const base::Value::List& args) {
   CHECK_EQ(args.size(), 2U);
-  PrefService* prefs = Profile::FromWebUI(web_ui())->GetPrefs();
   AllowJavascript();
+  brave_wallet::RemoveCustomNetwork(GetPrefs(), args[1].GetString());
+  ResolveJavascriptCallback(args[0], base::Value(true));
+}
+
+void BraveWalletHandler::ResetEthereumChain(const base::Value::List& args) {
```
Comment (yrliou) [inline browser/ui/webui/settings/brave_wallet_handler.cc:74]: Add a unit test for ResetEthereumChain in brave_wallet_handler_unittest.cc?

File: browser/ui/webui/settings/brave_wallet_handler.cc (hunk 2 above)
Comment (supermassive) [reply browser/ui/webui/settings/brave_wallet_handler.cc]: https://github.com/brave/brave-core/pull/13644/commits/a14bfe8d849769e50ad78160fe14bd5137bd9aef

File: browser/ui/webui/settings/brave_wallet_handler.cc (hunk 3 above)
Comment (supermassive) [reply browser/ui/webui/settings/brave_wallet_handler.cc:74]: https://github.com/brave/brave-core/pull/13644/commits/a14bfe8d849769e50ad78160fe14bd5137bd9aef

**Review by yrliou — APPROVED** [review]

**Pull request merged**